include/grpcpp/impl/codegen/server_callback.h

```diff
@@ -167,31 +172,103 @@ class ServerCallbackReaderWriter {
   }
 };
 
-// The following classes are reactors that are to be implemented
-// by the user, returned as the result of the method handler for
-// a callback method, and activated by the call to OnStarted
+// The following classes are the reactor interfaces that are to be implemented
+// by the user, returned as the result of the method handler for a callback
+// method, and activated by the call to OnStarted. Note that none of the classes
+// are pure; all reactions have a default empty reaction so that the user class
+// only needs to override those classes that it cares about.
+
+/// \a ServerBidiReactor is the interface for a bidirectional streaming RPC.
 template <class Request, class Response>
 class ServerBidiReactor : public internal::ServerReactor {
  public:
   ~ServerBidiReactor() = default;
-  virtual void OnStarted(ServerContext*) {}
+
+  /// Notify the application that a streaming RPC has started
+  ///
+  /// \param[in] context The context object now associated with this RPC
+  virtual void OnStarted(ServerContext* context) {}
+
+  /// Notifies the application that an explicit SendInitialMetadata operation
+  /// completed. Not used when the sending of initial metadata piggybacks onto
+  /// the first write.
+  ///
+  /// \param[in] ok Was it successful? If false, no further write-side operation
+  ///               will succeed
   virtual void OnSendInitialMetadataDone(bool ok) {}
+
+  /// Notifies the application that a Read operation completed.
+  ///
+  /// \param[in] ok Was it successful? If false, no further read-side operation
+  ///               will succeed
   virtual void OnReadDone(bool ok) {}
+
+  /// Notifies the application that a Write operation completed.
+  ///
+  /// \param[in] ok Was it successful? If false, no further read-side operation
+  ///               will succeed
   virtual void OnWriteDone(bool ok) {}
 
+  /// Send any initial metadata stored in the RPC context. If not invoked,
+  /// any initial metadata will be passed along with the first Write or the
+  /// Finish (if there are no writes).
   void StartSendInitialMetadata() { stream_->SendInitialMetadata(); }
+
+  /// Initiate a read operation.
+  ///
+  /// \param[out] msg Where to eventually store the read message. Valid when
```

Comment:
Use param names (msg, req, resp) uniformly throughout client and server APIs.

Reply:
Done!